### PR TITLE
Basic project setup

### PR DESCRIPTION
**Pull request by rpoltorak**
Please be aware of three things:

1. This is only for a kick-off. We need to start from somewhere. Any comments or propositions are more than welcome.
2. That basic CNN model is only for testing and demonstration purposes so the whole architecture and other values like image size, batch size, learning rate should be parameterized in the future (args)
3. It would be great to adjust the project structure in the future, so as to be able to easily switch between multiple models (and maybe datasets as well?)

### CODE REVIEW

File: mask_detector/data_prep.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def prepare_data():
+    df = pd.DataFrame()
+
+    for category in CATEGORIES:
+        path = DATASET_PATH/category
+
+        for directory in tqdm(list(path.iterdir())):
+            for imgPath in directory.iterdir():
```
Comment (GrivIN) [inline mask_detector/data_prep.py:18]: add check for file type (is an image?)

Comment (GrivIN) [reply mask_detector/data_prep.py:18]: imgPath -> img_path

File: mask_detector/data_prep.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                df = df.append({
+                    'image': str(imgPath),
+                    'mask': CATEGORIES.index(category)  # 0 - no_mask, 1 - mask
+                }, ignore_index=True)
```
Comment (GrivIN) [inline mask_detector/data_prep.py:22]: not sure but it may be slow, consider adding all dicts to an list and then create DF from that list in one step (append is copying whole df over and over)

File: mask_detector/data_prep.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import pandas as pd
+import os
+
+from pathlib import Path
+from tqdm import tqdm
```
Comment (GrivIN) [inline mask_detector/data_prep.py:5]: import ordering:

1. import builtin
2. from builtin
3. import third-party
4. from third-party

```suggestion
import os
from pathlib import Path

import pandas as pd
from tqdm import tqdm
```

File: mask_detector/data_prep.py (hunk 1 above)
Comment (GrivIN) [reply mask_detector/data_prep.py:18]: I am not convinced to var names used here path, directory seems too generic to me

File: mask_detector/data_prep.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    df = pd.DataFrame()
+
+    for category in CATEGORIES:
+        path = DATASET_PATH/category
+
+        for directory in tqdm(list(path.iterdir())):
+            for imgPath in directory.iterdir():
+                df = df.append({
+                    'image': str(imgPath),
+                    'mask': CATEGORIES.index(category)  # 0 - no_mask, 1 - mask
```
Comment (GrivIN) [inline mask_detector/data_prep.py:21]: do not use index on list for every loop, or move it outside the loop or use different datatype (like dict or set)

File: mask_detector/dataset.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import cv2
+import numpy as np
+import pandas as pd
+from torch import float, tensor
+from torch.utils.data.dataset import Dataset
+from torchvision.transforms import Compose, Resize, ToPILImage, ToTensor
+
+
+class MaskDataset(Dataset):
+    def __init__(self, csv_file, imageSize=32):
```
Comment (GrivIN) [inline mask_detector/dataset.py:12]: `imageSize` use sneak case

File: mask_detector/data_prep.py (hunk 1 above)
Comment (rpoltorak) [reply mask_detector/data_prep.py]: Great points! Thanks a lot. Fixed ;)

File: mask_detector/data_prep.py (hunk 2 above)
Comment (rpoltorak) [reply mask_detector/data_prep.py]: You're totally right. Fixed as well 👍

File: mask_detector/data_prep.py (hunk 3 above)
Comment (rpoltorak) [reply mask_detector/data_prep.py]: Definitely good practice, thanks!

File: mask_detector/data_prep.py (hunk 4 above)
Comment (rpoltorak) [reply mask_detector/data_prep.py]: Totally agree. Changed to dict & moved outside the loop as well.

**Comment by rpoltorak** [comment]
Switched pure pip with requirements.txt file to **pipenv**

**Review by GrivIN — COMMENTED** [review]
lgtm

**Review by GrivIN — APPROVED** [review]
